core/corehttp/gateway_handler.go

```diff
@@ -178,14 +178,14 @@ func (i *gatewayHandler) getOrHeadHandler(ctx context.Context, w http.ResponseWr
 		return
 	}
 
-	dr, err := i.api.Unixfs().Cat(ctx, resolvedPath)
+	dr, err := i.api.Unixfs().Get(ctx, resolvedPath)
 	dir := false
-	switch err {
-	case nil:
-		// Cat() worked
-		defer dr.Close()
-	case coreiface.ErrIsDir:
-		dir = true
+	switch {
+	case err == nil:
+		dir = dr.IsDirectory()
+		if !dir {
+			defer dr.Close()
```

Comment:
Actually... shouldn't we be closing this regardless? (I have no idea, this just looks weird).

Reply:
So technically for `go-ipfs-files.File`, in it's current form, `Close` should be only called for non-directories: https://github.com/ipfs/go-ipfs-files/blob/master/file.go#L18-L20

I'm working on refactoring files, making close valid for all files is one of the goals, but I wouldn't want to have this PR blocked on that refactor

Reply:
SGTM.